frontend/src/metabase-lib/fields.ts

```diff
@@ -30,3 +31,11 @@ export function fieldableColumns(
 export function _fieldId(column: ColumnMetadata): number | null {
   return ML.field_id(column);
 }
+
+export function findVisibleColumnForLegacyRef(
+  query: Query,
+  stageIndex: number,
+  fieldRef: FieldReference,
+): ColumnMetadata | null {
```

Comment:
I think we need a `ColumnWithOperators` from this, or maybe even more helpfully a way to turn a `ColumnMetadata` column into a `ColumnWithOperators` ?